Project settings fails to import when `externalOAuth` is set in the project

I was trying to import project settings using `terraform import ct_settings.project-definition my-project-key` but I got the following error:

```terraform
╷
│ Error: Request cancelled
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵


Stack trace from the terraform-provider-commercetools_v1.11.4 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 115 [running]:
github.com/labd/terraform-provider-commercetools/internal/resources/project.(*Project).setStateData(_, {{0x2, {0x140001861e0, 0x8}}, {0x0, {0x0, 0x0}}, {0x0, 0x0}, {0x0, ...}, ...})
        github.com/labd/terraform-provider-commercetools/internal/resources/project/model.go:120 +0x280
github.com/labd/terraform-provider-commercetools/internal/resources/project.(*ProjectResource).Read(0x14000636000, {0x103c44d00, 0x140006164e0}, {{{{0x103c49ae0, 0x140005d4db0}, {0x1039f0fc0, 0x140005d49f0}}, {0x103c4b3c8, 0x14000343770}}, 0x140001920f0, ...}, ...)
        github.com/labd/terraform-provider-commercetools/internal/resources/project/resource.go:336 +0x210
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x140001cd4a0, {0x103c44d00, 0x140006164e0}, 0x140006165d0, 0x140000c5400)
        github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwserver/server_readresource.go:101 +0x508
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ReadResource(0x140001cd4a0, {0x103c44d00?, 0x14000616390?}, 0x14000198000)
        github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/proto5server/server_readresource.go:56 +0x314
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ReadResource({0x140003ae540, 0x140003ae5a0, {0x14000371de0, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf5muxserver/mux_server_ReadResource.go:26 +0xdc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x140003b6b40, {0x103c44d00?, 0x140005261b0?}, 0x14000598600)
        github.com/hashicorp/terraform-plugin-go@v0.16.0/tfprotov5/tf5server/server.go:751 +0x3e8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x103bd1320?, 0x140003b6b40}, {0x103c44d00, 0x140005261b0}, 0x14000354000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.16.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:386 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002401e0, {0x103c49db8, 0x1400048a820}, 0x1400052e000, 0x1400047c8a0, 0x1043f53d0, 0x0)
        google.golang.org/grpc@v1.56.0/server.go:1337 +0xc90
google.golang.org/grpc.(*Server).handleStream(0x140002401e0, {0x103c49db8, 0x1400048a820}, 0x1400052e000, 0x0)
        google.golang.org/grpc@v1.56.0/server.go:1714 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.56.0/server.go:959 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.56.0/server.go:957 +0x16c

Error: The terraform-provider-commercetools_v1.11.4 plugin crashed!
```

The terraform config looks like this:
```terraform
resource "commercetools_project_settings" "project-definition" {
  key        = var.ct_project_key
  name       = var.ct_project_name
  countries  = ["GB"]
  currencies = ["GBP"]
  languages  = ["en-GB"]
  messages {
    enabled = false
  }
  carts {
    country_tax_rate_fallback_enabled   = false
    delete_days_after_last_modification = 90
  }
  enable_search_index_products = true
  enable_search_index_orders   = true
  external_oauth {
    url                  = var.ct_external_oauth_url
    authorization_header = "Bearer ${var.ct_external_oauth_secret}"
  }
}
```

The workaround was to remove from commercetools the externalOauth configuration 
```json
{
   ...
   "externalOAuth": {
        "url": "https://my-oauth-url",
        "authorizationHeader": "Bearer ****"
    },
    ...
```
and then I was able to import successfully.

--- a/.changes/unreleased/Fixed-20231115-124951.yaml
+++ b/.changes/unreleased/Fixed-20231115-124951.yaml
@@ -1,0 +1,4 @@
+kind: Fixed
+body: fixed check on project ExternalOAuth setting when processing state and plan
+  differences
+time: 2023-11-15T12:49:51.714555603+01:00

--- a/internal/resources/project/model.go
+++ b/internal/resources/project/model.go
@@ -116,7 +116,7 @@ func NewProjectFromNative(n *platform.Project) Project {
 }
 
 func (p *Project) setStateData(o Project) {
-	if len(p.ExternalOAuth) > 0 {
+	if len(p.ExternalOAuth) > 0 && len(o.ExternalOAuth) > 0 {
 		p.ExternalOAuth[0].AuthorizationHeader = o.ExternalOAuth[0].AuthorizationHeader
 	}
 
@@ -139,7 +139,7 @@ func (p *Project) setStateData(o Project) {
 	}
 }
 
-func (p Project) updateActions(plan Project) platform.ProjectUpdate {
+func (p *Project) updateActions(plan Project) platform.ProjectUpdate {
 	result := platform.ProjectUpdate{
 		Version: int(p.Version.ValueInt64()),
 		Actions: []platform.ProjectUpdateAction{},

--- a/internal/resources/project/model_test.go
+++ b/internal/resources/project/model_test.go
@@ -183,3 +183,75 @@ func TestUpdateActions(t *testing.T) {
 		})
 	}
 }
+
+func TestSetStateData(t *testing.T) {
+	tests := []struct {
+		name     string
+		state    Project
+		plan     Project
+		expected Project
+	}{
+		{
+			name: "externalOAuth nil",
+			state: Project{
+				ExternalOAuth: nil,
+				Carts: []Carts{
+					{},
+				},
+			},
+			plan: Project{
+				ExternalOAuth: nil,
+			},
+			expected: Project{
+				ExternalOAuth: nil,
+				Carts:         nil,
+			},
+		}, {
+			name: "externalOAuth in state",
+			state: Project{
+				ExternalOAuth: []ExternalOAuth{
+					{AuthorizationHeader: types.StringValue("some-value")},
+				},
+				Carts: []Carts{
+					{},
+				},
+			},
+			plan: Project{
+				ExternalOAuth: nil,
+			},
+			expected: Project{
+				ExternalOAuth: []ExternalOAuth{
+					{AuthorizationHeader: types.StringValue("some-value")},
+				},
+				Carts: nil,
+			},
+		}, {
+			name: "externalOAuth in plan",
+			state: Project{
+				ExternalOAuth: []ExternalOAuth{
+					{AuthorizationHeader: types.StringValue("some-value")},
+				},
+				Carts: []Carts{
+					{},
+				},
+			},
+			plan: Project{
+				ExternalOAuth: []ExternalOAuth{
+					{AuthorizationHeader: types.StringValue("some-other-value")},
+				},
+			},
+			expected: Project{
+				ExternalOAuth: []ExternalOAuth{
+					{AuthorizationHeader: types.StringValue("some-other-value")},
+				},
+				Carts: nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.state.setStateData(tt.plan)
+			assert.Equal(t, tt.expected, tt.state)
+		})
+	}
+}

